### PR TITLE
config: validate task messages and outputs

### PR DESCRIPTION
**Pull request by oliver-sanders**
At this moment users can do things like this:

```
[runtime]
  [[task]]
    [[[outputs]]]
      succeeded = failed  # shadowing builtin output, undefined behaviour
      a:b = abc  # not compatible with graph syntax
      required = required  # conflicts with cylc terminology
      BE $$ EF ££ WELL )(*& TON =  ?  # just nasty
```

* Task messages & outputs should be validated to avoid users shadowing built-in messages, outputs or qualifiers.
* Reserve the `_cylc` prefix save we need it later.

Also reserve the outputs `required`, `optional` and `all` as keywords for future use by `cylc set`, see [cylc set proposal](https://github.com/cylc/cylc-admin/pull/173). This proposal isn't approved yet and won't be implemented until Cylc 8.3.0, however, we may as well reserve these keywords now as the proposal will not change their meaning.

I've checked workflows at our end and have found shockingly little usage of custom outputs and no obvious breakages as a result of these changes.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/617.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -59,6 +59,10 @@ enhancements to `cylc lint`:
 
 ### Fixes
 
+[#5605](https://github.com/cylc/cylc-flow/pull/5605) -
```
Comment (oliver-sanders) [inline CHANGES.md:62]:
```suggestion
[#5606](https://github.com/cylc/cylc-flow/pull/5606) -
```

File: tests/integration/validate/test_outputs.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+    We don't want users adding outputs that override built-in
+    outputs (e.g. succeeded, failed) or qualifiers (e.g. succeed-all).
+
+    We don't want users adding outputs that conflict with keywords e.g.
+    "required" or "all".
+    """
+    # test that each output validates correctly
+    for output in outputs:
+        print(output)
```
Comment (hjoliver) [inline tests/integration/validate/test_outputs.py:78]:
```suggestion
```

(Debugging?)

File: tests/integration/validate/test_outputs.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    We don't want users adding messages that override built-in outputs (e.g.
+    succeeded, failed). To avoid confusion it's best to prohibit outputs which
+    override built-in qualifiers (e.g. succeed-all) too.
+
+    There's a special use of the colon character which users need to conform
+    with too.
+    """
+    # test that each message validates correctly
+    for message in messages:
+        print(message)
```
Comment (hjoliver) [inline tests/integration/validate/test_outputs.py:145]:
```suggestion
```
(Debugging?)

**Review by hjoliver — APPROVED** [review]
V. nice 👍

File: CHANGES.md (hunk 1 above)
Comment (MetRonnie) [reply CHANGES.md:62]: Are you going to apply this?

Comment (oliver-sanders) [reply CHANGES.md:62]: Yep, waiting for reviews.

File: cylc/flow/unicode_rules.py [hunk 4]
```diff
@@ -197,8 +237,10 @@ def not_equals(string):
 
     """
     return (
-        re.compile(rf'^(?!{re.escape(string)}$).*$'),
-        f'cannot be: ``{string}``'
+        re.compile(rf'^(?!^{_re_format_list(strings)}$).*$'),
+        # rf'^(?!{_re_format_list(strings)})$'),
+        formatted  _human_format_list([f'``{s}``' for s in strings])
+        f'cannot be: {formatted}'
```
Comment (MetRonnie) [inline cylc/flow/unicode_rules.py:243]: D'oh I made a mess of this the first time round
```suggestion
        'cannot be: ' + _human_format_list([f'``{s}``' for s in strings])
```

**Review by MetRonnie — CHANGES_REQUESTED** [review]

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**